toplevel/toploop.ml

```diff
@@ -140,10 +139,25 @@ let parse_mod_use_file name lb =
        ]
    ]
 
-(* Hooks for initialization *)
+(* Hook for initialization *)
 
 let toplevel_startup_hook = ref (fun () -> ())
 
+type event =
+  | Startup
+  | After_setup
```

Comment:
Would this be better as a polymorphic variant, or possibly extensible type, for (slightly) easier adding of future events?

Reply:
what would we gain by using a polymorphic variant? I'm not a big fan of them in general, except if you use their specific features

Reply:
I suppose yh. Let's use an open type, I don't really like over-using polymorphic variants as they produce awful error messages.